or_benchmarks/src/CheckerResultModule.cpp

```diff
@@ -0,0 +1,243 @@
+
+        start = std::chrono::high_resolution_clock::now();
+        check_result = checker_ptr -> CheckStandaloneSelfCollision(pbody);
+        end = std::chrono::high_resolution_clock::now();
+    }
+
+    else if (methodname == "CheckStandaloneSelfCollisionBaked")
+    {
+        OpenRAVE::KinBodyConstPtr pbody = DeserializeKinBody("body",check_log_val);
+
+        start = std::chrono::high_resolution_clock::now();
+        check_result = checker_ptr -> CheckStandaloneSelfCollision(pbody);
+        end = std::chrono::high_resolution_clock::now();
+    }
+
+    else
+    {
+        throw std::runtime_error("Unknown method - "+methodname);
+    }
+
+    std::chrono::duration<double> elapsed_seconds = end-start;
+    time_result = elapsed_seconds.count();
+
+    sout<<check_result<<" "<<time_result;
+
+
+    return true;
+
+}
+
+OpenRAVE::KinBodyConstPtr CheckerResultModule::DeserializeKinBody(std::string keyname, picojson::value check_log_val)
+{
+    std::string body_name(check_log_val.get(keyname).get<std::string>());
+    OpenRAVE::KinBodyConstPtr pbody = GetEnv() -> GetKinBody(body_name);
+    return pbody;
+}
+
+OpenRAVE::KinBody::LinkConstPtr CheckerResultModule::DeserializeLink(std::string keyname, picojson::value check_log_val)
+{
+    std::string link_and_body(check_log_val.get(keyname).get<std::string>());
+
+    size_t first_blank_pos = link_and_body.find(' ');
+
+    std::string link_name = link_and_body.substr(0,first_blank_pos);
+    std::string body_name = link_and_body.substr(first_blank_pos+1);
+
+    OpenRAVE::KinBodyConstPtr pbody = GetEnv() -> GetKinBody(body_name);
+    OpenRAVE::KinBody::LinkConstPtr plink = pbody -> GetLink(link_name);
+
+    return plink;
+}
+
+
+std::vector<OpenRAVE::KinBodyConstPtr> CheckerResultModule::DeserializeKinBodyList(std::string keyname, picojson::value check_log_val)
+{
+    picojson::value::array val_arr(check_log_val.get(keyname).get<picojson::value::array>());
+    std::vector<OpenRAVE::KinBodyConstPtr> vbodies;
+
+    for (auto val : val_arr)
+    {
+        OpenRAVE::KinBodyConstPtr pbody = GetEnv() -> GetKinBody(val.get<std::string>());
+        vbodies.push_back(pbody);
+    }
+
+    return vbodies;
+}
+
+
+std::vector<OpenRAVE::KinBody::LinkConstPtr> CheckerResultModule::DeserializeLinkList(std::string keyname, picojson::value check_log_val)
+{
+    picojson::value::array val_arr(check_log_val.get(keyname).get<picojson::value::array>());
+    std::vector<OpenRAVE::KinBody::LinkConstPtr> vlinks;
+
+    for (auto val : val_arr)
+    {
+        std::string link_and_body(val.get<std::string>());
+
+        size_t first_blank_pos = link_and_body.find(' ');
+
+        std::string link_name = link_and_body.substr(0,first_blank_pos);
+        std::string body_name = link_and_body.substr(first_blank_pos+1);
+
+        OpenRAVE::KinBodyConstPtr pbody = GetEnv() -> GetKinBody(body_name);
+        OpenRAVE::KinBody::LinkConstPtr plink = pbody -> GetLink(link_name);
```

Comment:
Check for `NULL` or, even better, replace this with a call to `DeserializeLink`.